apps/battery/app.js

```diff
@@ -105,236 +20,133 @@ module.exports = function (activity) {
           return
         }
         logger.log('notify media callback will exit:', url)
-        activity.exit()
+        this.exit()
       })
       .catch(error => {
         logger.warn(error)
       })
   }
 
-  function lowerPower (percent, isPlaying) {
-    var url
-    if (percent === 10) {
-      url = resourcePath.lowPower10
-    } else if (percent === 20) {
-      if (isPlaying && isPlaying === 'true') {
-        url = resourcePath.lowPower20Play
-      } else {
-        url = resourcePath.lowPower20Idle
-      }
+  function notifyTTS (text) {
+    if (text === false) {
+      return this.exit()
     }
-    notifyMedia(url)
+    logger.log('notifyTTS', text)
+    activity.tts.speak(text)
+      .then(() => {
+        logger.log('notify tts callback will exit:', text)
+        this.exit()
```

Comment:
Would preferring plain param `ctx` than `this` binding for much clear parameter dependency declararion.